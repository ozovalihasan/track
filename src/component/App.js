import React from 'react';
import { Route, Switch } from 'react-router-dom';
import ErrorContainer from './Error/ErrorContainer';
import LoadingContainer from './Loading/LoadingContainer';
import ListTrackedItemsContainer from './ListTrackedItems/ListTrackedItemsContainer';
import OneTrackedItemContainer from './OneTrackedItem/OneTrackedItemContainer';
import OnePieceContainer from './OnePiece/OnePieceContainer';
// import Header from '../component/Header';

const App = () => (
  <div className="App">
    <ErrorContainer />
    <LoadingContainer />

    {/* <div>
        <Header />
      </div> */}

    <Switch>
      <Route exact path="/" component={ListTrackedItemsContainer} />
      <Route exact path="/one-tracked-item/:id" component={OneTrackedItemContainer} />
      <Route exact path="/one-piece/:id" component={OnePieceContainer} />
    </Switch>
  </div>
);

export default App;
