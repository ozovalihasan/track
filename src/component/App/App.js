import React from 'react';
import { Route, Switch } from 'react-router-dom';
import styled from 'styled-components';
import AddCheckContainer from '../AddCheck/AddCheckContainer';
import ErrorContainer from '../Error/ErrorContainer';
import LoadingContainer from '../Loading/LoadingContainer';
import ListTakenTimesContainer from '../ListTakenTimes/ListTakenTimesContainer';
import ListTrackedItemsContainer from '../ListTrackedItems/ListTrackedItemsContainer';
import Menu from '../Menu/Menu';
import PieceFormContainer from '../PieceForm/PieceFormContainer';
import ProfileContainer from '../Profile/ProfileContainer';
import ShortTakenTimeContainer from '../ShortTakenTime/ShortTakenTimeContainer';
import ToggleShowHideContainer from '../ToggleShowHide/ToggleShowHideContainer';
import YourProgressContainer from '../YourProgress/YourProgressContainer';

const App = () => (
  <div className="App">

    <Main>
      <ErrorContainer />
      <LoadingContainer />

      <ToggleShowHideContainer>
        <ListTrackedItemsContainer />
      </ToggleShowHideContainer>

      <Switch>
        <Route exact path="/" component={AddCheckContainer} />
        <Route exact path="/one-piece-create" component={PieceFormContainer} />
        <Route exact path="/your-progress" component={YourProgressContainer} />
        <Route exact path="/profile" component={ProfileContainer} />
        <Route
          exact
          path="/list-taken-times"
          render={() => (
            <ListTakenTimesContainer>
              <ShortTakenTimeContainer />
            </ListTakenTimesContainer>
          )}
        />
      </Switch>
      <Menu />
    </Main>

  </div>
);

const Main = styled.div`
display: grid;
height: 100vh;
grid-template-rows: 10% 80% 10%;
`;

export default App;
