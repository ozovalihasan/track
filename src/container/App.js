import React from 'react';
// import { useSelector } from 'react-redux';
// import { Route, Switch } from 'react-router-dom';
import Loading from '../component/Loading';
import ErrorContainer from './ErrorContainer';
// import PokemonsListContainer from './PokemonsListContainer';
// import OnePokemonContainer from './OnePokemonContainer';
// import Header from '../component/Header';

function App() {
//   const loading = useSelector(state => state.pokemon.loading);
//   const error = useSelector(state => state.pokemon.error);

  return (
    <div className="App">
      <Loading />
      <ErrorContainer />
      {/* <div>
        <Header />
      </div> */}
      {/* {loading && <Loading />}
      {error && <ErrorContainer />}
      <Switch>
        <Route exact path="/" component={PokemonsListContainer} />
        <Route exact path="/one-pokemon/:id" component={OnePokemonContainer} />
      </Switch> */}
    </div>
  );
}

export default App;
