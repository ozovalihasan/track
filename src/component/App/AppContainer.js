import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { fetchUserAutoLogin } from '../../redux';
import App from './App';

const AppContainer = () => {
  const showList = useSelector(state => state.app.showList);

  const dispatch = useDispatch();
  useEffect(() => {
    if (localStorage.token) dispatch(fetchUserAutoLogin());
  }, []);

  const user = useSelector(state => state.user);

  if (user.user && localStorage.token) {
    return (<App showList={showList} />);
  }

  return (
    <Redirect to={{ pathname: '/login' }} />
  );
};

export default AppContainer;
