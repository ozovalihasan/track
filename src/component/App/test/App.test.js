import { render, screen } from '@testing-library/react';
import React from 'react';
import PropTypes from 'prop-types';
import {
  BrowserRouter, Redirect, Route, Switch,
} from 'react-router-dom';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import App from '../App';

jest.mock('../../AddCheck/AddCheckContainer', () => {
  const AddCheckContainer = () => (<div>Mock AddCheckContainer </div>);
  AddCheckContainer.displayName = 'AddCheckContainer';
  return AddCheckContainer;
});

jest.mock('../../Error/ErrorContainer', () => {
  const ErrorContainer = () => (<div>Mock ErrorContainer </div>);
  ErrorContainer.displayName = 'ErrorContainer';
  return ErrorContainer;
});

jest.mock('../../Loading/LoadingContainer', () => {
  const LoadingContainer = () => (<div>Mock LoadingContainer </div>);
  LoadingContainer.displayName = 'LoadingContainer';
  return LoadingContainer;
});

jest.mock('../../ListTakenTimes/ListTakenTimesContainer', () => {
  const ListTakenTimesContainer = () => (<div>Mock ListTakenTimesContainer </div>);
  ListTakenTimesContainer.displayName = 'ListTakenTimesContainer';
  return ListTakenTimesContainer;
});

jest.mock('../../ListTrackedItems/ListTrackedItemsContainer', () => {
  const ListTrackedItemsContainer = () => (<div>Mock ListTrackedItemsContainer </div>);
  ListTrackedItemsContainer.displayName = 'ListTrackedItemsContainer';
  return ListTrackedItemsContainer;
});

jest.mock('../../Menu/Menu', () => {
  const Menu = () => (<div>Mock Menu </div>);
  Menu.displayName = 'Menu';
  return Menu;
});

jest.mock('../../PieceForm/PieceFormContainer', () => {
  const PieceFormContainer = () => (<div>Mock PieceFormContainer </div>);
  PieceFormContainer.displayName = 'PieceFormContainer';
  return PieceFormContainer;
});

jest.mock('../../Profile/ProfileContainer', () => {
  const ProfileContainer = () => (<div>Mock ProfileContainer </div>);
  ProfileContainer.displayName = 'ProfileContainer';
  return ProfileContainer;
});

jest.mock('../../ShortTakenTime/ShortTakenTimeContainer', () => {
  const ShortTakenTimeContainer = () => (<div>Mock ShortTakenTimeContainer </div>);
  ShortTakenTimeContainer.displayName = 'ShortTakenTimeContainer';
  return ShortTakenTimeContainer;
});

jest.mock('../../ToggleShowHide/ToggleShowHideContainer', () => {
  const ToggleShowHideContainer = () => (<div>Mock ToggleShowHideContainer</div>);
  ToggleShowHideContainer.displayName = 'ToggleShowHideContainer';
  return ToggleShowHideContainer;
});

jest.mock('../../YourProgress/YourProgressContainer', () => {
  const YourProgressContainer = () => (<div>Mock YourProgressContainer </div>);
  YourProgressContainer.displayName = 'YourProgressContainer';
  return YourProgressContainer;
});

describe('<App />', () => {
  it('renders ErrorContainer,LoadingContainer,ToggleShowHideContainer, and Menu components', () => {
    render(
      <BrowserRouter>
        <App />
        {/* <Redirect to={{ pathname: '/one-piece-create' }} /> */}
      </BrowserRouter>,
    );

    expect(screen.getByText(/Mock ErrorContainer/i)).toBeInTheDocument();
    expect(screen.getByText(/Mock LoadingContainer/i)).toBeInTheDocument();
    expect(screen.getByText(/Mock ToggleShowHideContainer/i)).toBeInTheDocument();
    expect(screen.getByText(/Mock Menu/i)).toBeInTheDocument();
  });

  it('renders AddCheckContainer component when \'/\' is directed', () => {
    render(
      <BrowserRouter>
        <App />
        <Redirect to={{ pathname: '/' }} />
      </BrowserRouter>,
    );

    expect(screen.getByText(/Mock AddCheckContainer/i)).toBeInTheDocument();
  });

  it('renders PieceFormContainer component when \'/one-piece-create\' is directed', () => {
    render(
      <BrowserRouter>
        <App />
        <Redirect to={{ pathname: '/one-piece-create' }} />
      </BrowserRouter>,
    );

    expect(screen.getByText(/Mock PieceFormContainer/i)).toBeInTheDocument();
  });

  it('renders YourProgressContainer component when \'/your-progress\' is directed', () => {
    render(
      <BrowserRouter>
        <App />
        <Redirect to={{ pathname: '/your-progress' }} />
      </BrowserRouter>,
    );

    expect(screen.getByText(/Mock YourProgressContainer/i)).toBeInTheDocument();
  });

  it('renders ProfileContainer component when \'/profile\' is directed', () => {
    render(
      <BrowserRouter>
        <App />
        <Redirect to={{ pathname: '/profile' }} />
      </BrowserRouter>,
    );

    expect(screen.getByText(/Mock ProfileContainer/i)).toBeInTheDocument();
  });

  it('renders ListTakenTimesContainer component when \'/list-taken-times\' is directed', () => {
    render(
      <BrowserRouter>
        <App />
        <Redirect to={{ pathname: '/list-taken-times' }} />
      </BrowserRouter>,
    );

    expect(screen.getByText(/Mock ListTakenTimesContainer/i)).toBeInTheDocument();
  });

  it('renders correctly', () => {
    const renderedContainer = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );
    expect(renderedContainer).toMatchSnapshot();
  });
});
