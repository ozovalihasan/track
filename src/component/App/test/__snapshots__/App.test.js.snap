// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> If there is a user If there is a tracked item renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          Mock ErrorContainer 
        </div>
        <div>
          Mock LoadingContainer 
        </div>
        <div
          class="sc-bdfBwQ eSkTZp"
        >
          <div>
            Mock ToggleShowHideContainer
          </div>
          <div>
            Mock AddCheckContainer 
          </div>
        </div>
        <div>
          Mock MenuContainer 
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        Mock ErrorContainer 
      </div>
      <div>
        Mock LoadingContainer 
      </div>
      <div
        class="sc-bdfBwQ eSkTZp"
      >
        <div>
          Mock ToggleShowHideContainer
        </div>
        <div>
          Mock AddCheckContainer 
        </div>
      </div>
      <div>
        Mock MenuContainer 
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<App /> If there is a user If there is no any tracked item renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          Mock ErrorContainer 
        </div>
        <div>
          Mock LoadingContainer 
        </div>
        <div>
          Mock ListTrackedItemsContainer 
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        Mock ErrorContainer 
      </div>
      <div>
        Mock LoadingContainer 
      </div>
      <div>
        Mock ListTrackedItemsContainer 
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<App /> If there is no any user renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          Mock ErrorContainer 
        </div>
        <div>
          Mock LoadingContainer 
        </div>
        <div>
          Mock LogIn 
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        Mock ErrorContainer 
      </div>
      <div>
        Mock LoadingContainer 
      </div>
      <div>
        Mock LogIn 
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
